uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequestComment.php

```diff
@@ -11,37 +11,20 @@ class CRM_HRLeaveAndAbsences_Service_LeaveRequestComment {
   private $entityName = 'LeaveRequest';
 
   /**
-   * @var array
-   *   The parameter array needed to add, create or delete comments
-   */
-  private $params;
-
-  /**
-   * CRM_HRLeaveAndAbsences_Service_LeaveRequestComment constructor.
-   *
-   * @param $params
-   *   The parameter array needed to add, create or delete comments
-   */
-  public function __construct($params) {
-    $this->params = $params;
-    $this->prepareParametersForCommentPayload();
-  }
-
-  /**
    * Uses the API exposed by the Comment entity (Comment.create)
    * to create comments related to a LeaveRequest.
    *
    * @throws UnexpectedValueException
    *
    * @return array
    */
-  public function add() {
+  public function add($params) {
```

Comment:
please add $params to the dockblock of the method (also applies to the other methods)